GRDB/Core/SerializedDatabase.swift

```diff
@@ -42,7 +42,12 @@ final class SerializedDatabase {
         self.queue = configuration.makeDispatchQueue(defaultLabel: defaultLabel, purpose: purpose)
         SchedulingWatchdog.allowDatabase(db, onQueue: queue)
         try queue.sync {
-            try db.setup()
+            do {
+                try db.setup()
+            } catch {
+                db.close()
```

Comment:
Please take a close look at this change. 👀 

We used to call `Database.closeConnection` in Database.init, but now all the `throw`able things from Database.init that would require closing the database have been moved to `db.setup`

Reply:
Yes, this looks correct, too 👍